### PR TITLE
GitHub Action to build and publish Docker image

### PR DESCRIPTION
**Pull request by mkindahl**
This introduces a GitHub Action to build and publish a TimescaleDB
docker image. It is a copy of the existing method for building docker
images.

### CODE REVIEW

File: docker-build-and-publish/action.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Build and Publish TimescaleDB Docker Image
+description: Build a TimescaleDB Docker image ready for publishing
+inputs:
```
Comment (cevian) [inline docker-build-and-publish/action.yml:3]: should we have an optional input for push? You may want to build an image without pushing (for testing the build process)

File: docker-build-and-publish/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,86 @@
+ARG PG_VERSION
```
Comment (cevian) [inline docker-build-and-publish/Dockerfile:1]: not positive this should be in here or in the repo with the code and provided in the workspace

**Review by cevian — APPROVED** [review]

File: docker-build-and-publish/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+ARG PREV_TS_VERSION=2.3.0
+ARG PREV_EXTRA
+############################
+# Build tools binaries in separate image
+############################
+ARG GO_VERSION=1.14.0
+FROM golang:${GO_VERSION}-alpine AS tools
+
+ENV TOOLS_VERSION 0.8.1
```
Comment (mkindahl) [inline docker-build-and-publish/Dockerfile:10]: Might be good to expose this as a parameter to the action as well.

File: docker-build-and-publish/Dockerfile (hunk 2 above)
Comment (mkindahl) [reply docker-build-and-publish/Dockerfile:1]: I didn't follow that. What alternative do you propose?

File: docker-build-and-publish/action.yml (hunk 1 above)
Comment (mkindahl) [reply docker-build-and-publish/action.yml:3]: Added one, and one for the tools version. Not particularly fond of the `Dockerfile`, but we can refactor and optimize later.

File: docker-build-and-publish/Dockerfile (hunk 3 above)
Comment (mkindahl) [reply docker-build-and-publish/Dockerfile:10]: Added it as a parameter to the action, with this as default.

File: docker-build-and-publish/Dockerfile (hunk 2 above)
Comment (cevian) [reply docker-build-and-publish/Dockerfile:1]: The dockerfile could be in the timescaleDB repo passed in via the workspace. That way you only need to provide the path to the dockerfile in the action

**Pull request merged**